portal-backend/depmap/data_explorer_2/views.py

```diff
@@ -462,18 +463,25 @@ def get_labels_matching_context():
     return make_gzipped_json_response(labels_matching_context)
 
 
+# TODO: Remove this endpoint. It's only used for one specific feature type
+# which is compound_experiment. The DE2 UI has a special case for that type
+# where it allows you to start by searching on a compound name alone. Then the
+# dataset name is used to resolve ambiguities in the relatively small number of
+# cases where there's more than one experiment per compound. In the future, we
+# can solve this a manner more consistent with other types by leveraging the
+# search indexer.
```

Comment:
Also, since I'm trying to keep track of everything that needs to be done for Elara V2, I added a task for this to Asana: https://app.asana.com/0/1165651979405609/1208160165634335/f